build.assets/Makefile

```diff
@@ -125,7 +128,7 @@ build-binaries-fips: buildbox-centos7-fips webassets
 .PHONY:buildbox
 buildbox:
 	$(call PULL_ON_CI,$(BUILDBOX))
-	DOCKER_BUILDKIT=1 docker build --platform=linux/$(RUNTIME_ARCH) \
+	DOCKER_BUILDKIT=1 docker build --platform $(OS)/$(ARCH) \
```

Comment:
1. I don't think that ubuntu os centos are released on OS != linux. This will probably break builds on macOS (I can be wrong on that).
2. There is a reason why we have ARCH and RUNTIME_ARCH. When you're on a M1 Mac and want to build x86 binaries it used to be the case that `ARCH=amd64 make release` used to cross-compile x86 binaries on arm64 docker image. From what I see this functionality is broken. I'd ask in dev channel if anyone uses it. If not I'd just remove the RUNTIME_ARCH.

That being said, I think we should remove the `--platform` at all from all our `docker build`. The virtualization never worked for us (that's why we build ARM on ARM CPU) and having this option here just make it confusing.

Reply:
1. There's no `darwin/arm64` for Ubuntu (expected), but `OS=linux make -C build.assets buildbox` works fine on my M1 Mac (with this branch). We could hardcode it as `linux` by default but allow for overwriting. I can add that.

2. Our use of `ARCH` and `RUNTIME_ARCH` is messy. I'm fine supporting the functionality you mention.

This is what we have now in `build.assets/Makefile`:

```
ARCH ?= $(shell go env GOARCH)
HOST_ARCH := $(shell uname -m)
RUNTIME_ARCH_x86_64 := amd64
# uname returns different value on Linux (aarch64) and macOS (arm64).
RUNTIME_ARCH_arm64 := arm64
RUNTIME_ARCH_aarch64 := arm64
RUNTIME_ARCH := $(RUNTIME_ARCH_$(HOST_ARCH))
```

Should make sure we understand the differences between the three (or consolidate as appropriate).

Reply:
> What has been the problem with --platform? 

I think that the current problem is the amount of custom logic we have around it. We started from a simple idea or having a cross-compilation, from what I recall. Then we added the `--platform` to have compilation in a virtualized docker. Currently, we have 4 or 5 different architecture-related flags, and no one knows what they are supposed to do. 

> OS=linux make -C build.assets buildbox works fine on my M1 Mac (with this branch)

This build always worked, as it builds arm64 version of the Ubuntu image. Docker by default builds a native image, so in case of M1, that will be `linux/arm64`.

Here is my proposal. Let's remove all ARCH-related flags for now. The only build that still needs to work is `make grpc` that generates the protos on amd64 linux and M1 Mac. Then, if we want to, we can add support for the `--platform` switch. We can just let override it `--platform=linux/$(DOCKER_PLATFROM_ARCH)`.

My personal complains about `--platform` are:
1. It's slow. Sometimes very slow.
2. Our centos7 image doesn't build with that flag. We should probably fix that if we want to support amd64 builds on M1.

Reply:
I ripped out all the `--platform` stuff. We can add it back later.

Reply:
As a quick note, `make grpc` doesn't care about platform and it should just work, given that is a very simple image.